### PR TITLE
[do not merge] [tests] tidy up wallet_importmulti.py

### PR DESCRIPTION
**Pull request by jnewbery**
Based on #14565. Please don't review this until that PR is merged.

Fixes review comments from @ryanofsky here: https://github.com/bitcoin/bitcoin/pull/14886#pullrequestreview-183772779

### CODE REVIEW

File: doc/release-notes-14565.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Low-level RPC changes
+---------------------
+
+The `importmulti` RPC will now contain a new per-request `warnings` field with strings
+that explain when fields are being ignored or inconsistant, if any.
```
Comment (practicalswift) [inline doc/release-notes-14565.md]: Should be "inconsistent" :-)

File: test/functional/wallet_importmulti.py [hunk 2]
```diff
@@ -80,63 +44,18 @@ def skip_test_if_missing_module(self):
...
-                        CScript([OP_HASH160, witness_script, OP_EQUAL]).hex(),  # p2sh-p2wsh
-                        script_to_p2sh_p2wsh(script_code))  # p2sh-p2wsh addr
-
-    def test_importmulti(self, req, success, error_code=None, error_message=None):
+    def test_importmulti(self, req, success, error_code=None, error_message=None, warnings=[]):
         """Run importmulti and assert success"""
         result = self.nodes[1].importmulti([req])
+        observed_warnings = []
+        if 'warnings' in result[0]:
+           observed_warnings = result[0]['warnings']
```
Comment (practicalswift) [inline test/functional/wallet_importmulti.py]: Indentation is not a multiple of four :-)

File: test/functional/wallet_importmulti.py [hunk 3]
```diff
@@ -80,63 +44,18 @@ def skip_test_if_missing_module(self):
...
-                        CScript([OP_HASH160, hash160(script_code), OP_EQUAL]).hex(),  # p2sh
-                        script_to_p2sh(script_code),  # p2sh addr
-                        script_code.hex(),  # redeem script
-                        witness_script.hex(),  # p2wsh
-                        script_to_p2wsh(script_code),  # p2wsh addr
-                        CScript([OP_HASH160, witness_script, OP_EQUAL]).hex(),  # p2sh-p2wsh
-                        script_to_p2sh_p2wsh(script_code))  # p2sh-p2wsh addr
-
-    def test_importmulti(self, req, success, error_code=None, error_message=None):
+    def test_importmulti(self, req, success, error_code=None, error_message=None, warnings=[]):
```
Comment (practicalswift) [inline test/functional/wallet_importmulti.py]: Do you want `test_importmulti` to remember warnings by using `warnings=[]`? If so, add a comment :-)

```
>>> def test(i, i_arr=[]):
...     i_arr.append(i)
...     return i_arr
...
>>> test(1)
[1]
>>> test(2)
[1, 2]
>>> test(3)
[1, 2, 3]
```

**Comment by jnewbery** [comment]
@practicalswift - please see the PR text:

> Based on #14565. Please don't review this until that PR is merged.

Two of your three review comments above are for commits in #14565.

**Comment by Sjors** [comment]
Meanwhile, concept ACK on `wallet_util.py`

**Comment by MarcoFalke** [comment]
Picked up in #15108

**Comment by jnewbery** [comment]
@amitiuttarwar has offered to pick this up in #15108. Reviewers, please leave review comments there.

**Pull request closed without merging**